examples/RodContactCase/post_processing.py

```diff
@@ -0,0 +1,375 @@
+    FFMpegWriter = animation.writers["ffmpeg"]
+    metadata = dict(title="Movie Test", artist="Matplotlib", comment="Movie support!")
+    writer = FFMpegWriter(fps=fps, metadata=metadata)
+    dpi = kwargs.get("dpi", 100)
+
+    xlim = kwargs.get("x_limits", (-1.0, 1.0))
+    ylim = kwargs.get("y_limits", (-1.0, 1.0))
+    zlim = kwargs.get("z_limits", (-0.05, 1.0))
+
+    difference = lambda x: x[1] - x[0]
+    max_axis_length = max(difference(xlim), difference(ylim))
+    # The scaling factor from physical space to matplotlib space
+    scaling_factor = (2 * 0.1) / max_axis_length  # Octopus head dimension
+    scaling_factor *= 2.6e3  # Along one-axis
+
+    if kwargs.get("vis3D", True):
+        fig = plt.figure(1, figsize=(10, 8), frameon=True, dpi=dpi)
+        ax = plt.axes(projection="3d")
+
+        ax.set_xlabel("x")
+        ax.set_ylabel("y")
+        ax.set_zlabel("z")
+
+        ax.set_xlim(*xlim)
+        ax.set_ylim(*ylim)
+        ax.set_zlim(*zlim)
+
+        time_idx = 0
+        rod_lines = [None for _ in range(n_visualized_rods)]
+        rod_com_lines = [None for _ in range(n_visualized_rods)]
+        rod_scatters = [None for _ in range(n_visualized_rods)]
+
+        for rod_idx in range(n_visualized_rods):
+            inst_position, inst_radius = rod_history_unpacker(rod_idx, time_idx)
+            if not inst_position.shape[1] == inst_radius.shape[0]:
+                inst_position = 0.5 * (inst_position[..., 1:] + inst_position[..., :-1])
+
+            rod_scatters[rod_idx] = ax.scatter(
+                inst_position[0],
+                inst_position[1],
+                inst_position[2],
+                s=np.pi * (scaling_factor * inst_radius) ** 2,
+            )
+
+        # ax.set_aspect("equal")
+        video_name_3D = "3D_" + video_name
+
+        with writer.saving(fig, video_name_3D, dpi):
+            with plt.style.context("seaborn-whitegrid"):
+                for time_idx in tqdm(range(0, sim_time.shape[0], int(step))):
+
+                    for rod_idx in range(n_visualized_rods):
+                        inst_position, inst_radius = rod_history_unpacker(
+                            rod_idx, time_idx
+                        )
+                        if not inst_position.shape[1] == inst_radius.shape[0]:
+                            inst_position = 0.5 * (
+                                    inst_position[..., 1:] + inst_position[..., :-1]
+                            )
+
+                        rod_scatters[rod_idx]._offsets3d = (
+                            inst_position[0],
+                            inst_position[1],
+                            inst_position[2],
+                        )
+
+                        # rod_scatters[rod_idx].set_offsets(inst_position[:2].T)
+                        rod_scatters[rod_idx].set_sizes(
+                            np.pi * (scaling_factor * inst_radius) ** 2
+                        )
+
+                    writer.grab_frame()
```

Comment:
This is something I have seen pervade pyelastica. Is there a good reason the animation module is eschewed and frames are grabbed manually?

Reply:
I think there is no reason. It is just the practice, we keep doing. Do you have better suggestion? We can check animation module, I am not very familiar.

Reply:
IIRC you don't need manual control over frames (like 'grab_frame`) and figures (like `plt.close(plt.gcf()`) if you use the animation module . But this is not a huge issue. Its not library code anyway.